backend/src/main/java/hu/flowacademy/konyhatunder/service/CategoryService.java

```diff
@@ -25,8 +22,11 @@
         return categoryRepository.findAll();
     }
 
-    public Optional<Category> findById(String id) {
-        return categoryRepository.findById(id);
+    public Category findById(String id) {
+        Category category = categoryRepository.findById(id).orElse(null);
```

Comment:
Itt ket dolgot megjegyeznek:
1. Az egy soros if statement jo, es hasznaljuk nyugodtan, de legyen tenyleg 1 sor. Ezt igy foleg egy nagyobb methodban be lehet nezni. Itt viszont effektive amugy is egy kvazi else agba futnank, igy erdemesebb kitenni a bracketeket.
2. Ha mar Optionalt kapunk valahonnan akkor hasznaljuk. Optionallal, ha .orElse(null)-t irunk akkor valoszinuleg rosszul hasznaltuk az optionalt, es igazabol nem is volt ertelme hogy azt kaptunk vissza.

Itt a megoldas valami ilyesmi lenne:
return categoryRepository.findByid(id).orElseThrow(() -> new ValidationException("Nincs ilyen ID-val rendelkező kategória"));

Reply:
Tovabba, bar nem ide kapcsolodik, legalabb a constans stringeket tegyetek ki constans strigekbe, akar a classon belul, vagy egy ErrorMessageConstants classba. Ez utobbi azert elonyos mert bizonyos msg-ek ujra hasznalhatoak, es nem kell vadaszni rajuk ha at kell irni egyet, ami tobb helyen is szerepel.

Reply:
A visszetérést refaktoráltam. 
Konstans stringünk igazából szerintem nincs, mindenhol más szöveget adunk az exceptionben.

Reply:
Attol, hogy csak egy helyen hasznalod, ott meg mindig ugyan azt hasznalod. Egy beegetett konstanst hasznalsz, csak nincs kiszervezve valtozoba, ezeket hivjuk Magic String-eknek. Ez most csak Clean Code kukacoskodas. De ha esetleg szukseg lesz egy I18N megoldasra, vagy csak azt mondja a megrendelo, hogy legyen angol, akkor kiszervezett constansokkal egyszserubb dolgozni.